### DISCUSSION TITLE
How to share mutable states in .with_file_drop_handler?

### DISCUSSION THREAD

**Discussion by ZihuiLiang** [DioxusLabs/dioxus#1069] [Q&A]
I want to capture the dropped file by .with_file_drop_handler. Is there any way to share mutable states in the handler? It seems hooks cannot be used since an error information "borrowed data escapes outside of function".  Also I hope there is a good way to upload the file and share mutable states to record the information of uploaded files. It seems that I can only use .with_file_drop_handler and meet the previous problem. The example in https://github.com/DioxusLabs/dioxus/blob/master/examples/file_upload.rs fails on my computer.

**Comment by ZihuiLiang** [comment]
It seems that using the published dioxus packages may fail on implementing some examples. For example I used the latest published version of dioxus, dioxus-desktop and implemented the file_upload example. I found the app fails on uploading the files. Finally, I use the packages on the GitHub and succeed.

**Comment by ealmloff** [comment]
The published version just has a placeholder file handler trait.

If you want just the examples that should work with the published version, you can look at the 0.3.2 tag on GitHub: https://github.com/DioxusLabs/dioxus/tree/2075ff3039feb1efb576d8f2ed5a2c4d0c554731

**Reply by ZihuiLiang** [reply]
Indeed, initially, I used cargo to add the dioxus packages with the latest version (0.3.2). But applied those packages, the file upload example failed. Are these packages different from the ones on GitHub?

Also,sometimes, cargo update some dependencies of dioxus (memoffset,field-offset,cfg-expr, dioxus-interpreter-js) so that those dependencies made the program failed. In the end, I used their fixed versions
